### PR TITLE
Run tests on CI

### PR DESCRIPTION
**Pull request by hartsick**
* Run on every push for now, to test

#35

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      uses: ruby/setup-ruby@v1
+    - run: ruby -v
+    
+    - name: Setup Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '14.18.x'
+        registry-url: 'https://registry.npmjs.org'
+    
+    - name: Cache Gems
```
Comment (rahearn) [inline .github/workflows/test.yml:32]: [suggestion] The `Setup Ruby` step can do this for us. See https://github.com/OHS-Hosting-Infrastructure/complaint-tracker/blob/e9452643d5d672c47a94fe9e44b86bf09bcf33ef/.github/actions/setup-project/action.yml#L10 for an example.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      uses: actions/cache@v1.1.2
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: ${{ runner.os }}-gems-
+
+    - name: Install and configure bundler
```
Comment (rahearn) [inline .github/workflows/test.yml:39]: [suggestion] This step is also unnecessary if we have `setup-ruby` handle it.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      run: |
+        gem install bundler
+        bundle config path vendor/bundle
+        bundle install --jobs 4 --retry 3
+
+    - name: Yarn cache
```
Comment (rahearn) [inline .github/workflows/test.yml:45]: [suggestion] `setup-node` will cache by default, though not run `yarn install` for us. See https://github.com/OHS-Hosting-Infrastructure/complaint-tracker/blob/e9452643d5d672c47a94fe9e44b86bf09bcf33ef/.github/actions/setup-javascript/action.yml for an example

**Review by rahearn — CHANGES_REQUESTED** [review]
We can simplify this a bit by letting the setup steps handle dependency cacheing

**Comment by hartsick** [comment]
Closing since we have a lot of this infrastructure now! Will resubmit off latest main

**Pull request closed without merging**